[Koh Hong Po] iP

hpkoh opened the pull request:
# WingerIO
> Denny's is for winners

Turns out you are who winger's **always** on the phone with, he helps you
- [x] Manage Tasks
- [ ] Give reminders


Types of tasks currently  supported:
* Todo
* Deadline
* Event

Running it on a terminal
1. Download the jar file [here](https://github.com/hpkoh/ip/releases/tag/A-Jar)
2. Go to the directory where you saved the file
3. Run the command ```java -jar {filename}.jar```

This is how *Duke* class is ran 🏃 
```Java
public class Main {
    public static void main(String[] args) {
        new Duke("path/to/save/file").run();
    }
}
```

Commit:
Add Gradle support

hpkoh committed:
Level 1

Add Greeting/Exit message
Bot echoes user input

hpkoh committed:
Level-2

Add todo list
Format output string

User can add and view their todo list

hpkoh committed:
Level-3

Add mark as done feature
Completion of Tasks are tracked

hpkoh committed:
Level-4

Add Tasks: ToDo, Deadline, Event

hpkoh committed:
A-TextUiTesting

hpkoh committed:
Level-5

Add error handling for lack of description

hpkoh committed:
Level-6

Add delete feature

hpkoh committed:
Level-7

Add save file feature

hpkoh committed:
Level-8

Add LocalDate object support

hpkoh committed:
Merge branch 'branch-Level-7'

hpkoh committed:
Merge branch 'branch-Level-8'

hpkoh committed:
A-MoreOOP

Refractor to encapsulate Ui, Storage, Parser, TaskList, Command

hpkoh committed:
A-Packages

Refractor classes into commands, exceptions, tasks, utils packages

hpkoh committed:
A-JUnit

Add JUnit tests for Parser and Ui

hpkoh committed:
A-JavaDoc

hpkoh committed:
A-CodingStandard

hpkoh committed:
Level-9

Add find feature

hpkoh committed:
Merge branch 'branch-A-JavaDoc'

hpkoh committed:
Merge branch 'branch-A-CodingStandard'

hpkoh committed:
Merge branch 'branch-Level-9'

kwokieee commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,47 @@
+    public void run() {
+        ui.printWelcomeMessage();
+
+        boolean notExitCommand = true;
```
```suggestion
        boolean isExitCommand = false;
```

hpkoh replied:
isExitCommand does make it clearer! Thanks

kwokieee commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,47 @@
+                ui.printError(e.getMessage());
+            }          
+        }
+        ui.printExitMessage();
```
Should this be encapsulated within `ExitCommand::execute`?

hpkoh replied:
Nice catch! Execute handling the message does make more sense!

kwokieee commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+
+public class Deadline extends Task{
+
+    private LocalDate time;
```
```suggestion
    private LocalDate date;
```

hpkoh replied:
Since I'm using dates, naming it date does make more sense. Thanks!

kwokieee commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,38 @@
+
+public class Event extends Task{
+
+    private LocalDate startTime;
+    private LocalDate endTime;
```
```suggestion
    private LocalDate startDate;
    private LocalDate endDate;
```

kwokieee commented on src/main/java/utils/Parser.java:
```diff
@@ -0,0 +1,79 @@
+
+import java.time.LocalDate;
+
+import commands.*;
```
Be careful not to use wildcard imports!

hpkoh replied:
Thanks for the heads up!

kwokieee commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,27 @@
+package commands;
```
Would it be better if the root name of the package was the project name?
```suggestion
package duke.commands;
```

hpkoh replied:
Absolutely, great catch again!

kwokieee reviewed (commented):
Overall, I found that you followed the coding standard well! 😊 I noted a few minor violations to fix, and I have also made some suggestions to some names that may have been a bit misleading.

wyrchris commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,47 @@
+import utils.Parser;
+import utils.Storage;
+import utils.Ui;
+
```
Missing headers comments for public classes

hpkoh replied:
Header comments will definitely aid in clarity. Will add headers!

wyrchris commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,27 @@
+        this.task = task;
+    }
+
+    public boolean execute(TaskList tasks, Ui ui, Storage storage) {
```
Could include the annotation to alert reader that it is overriding the abstract method
```suggestion
    @Override
    public boolean execute(TaskList tasks, Ui ui, Storage storage) {
```

hpkoh replied:
Nice catch! thanks

wyrchris commented on src/main/java/utils/Parser.java:
```diff
@@ -0,0 +1,79 @@
+
+        String description = commands.length > 1 ? commands[1] : "";
+        
+        if (command.equals("bye")) {
```
Could consider using a switch-case statement for greater clarity

hpkoh replied:
I agree! Will implement alongside enums.

wyrchris commented on src/main/java/utils/Storage.java:
```diff
@@ -0,0 +1,77 @@
+            while (dataReader.hasNextLine()) {
+                String[] data = dataReader.nextLine().split("\\|");
+                String type = data[0];
+                boolean isDone = data[1] == "1" ? true : false;
```
Need not explictly return true/false
```suggestion
                boolean isDone = data[1] == "1";
```

hpkoh replied:
Nice catch! Thanks

wyrchris reviewed (commented):
Other than including comments/Javadocs to the relevant class/method, overall the code looks structured well and is easy to understand!

hpkoh commented:
# WingerIO
> Denny's is for winners

Turns out you are who winger's **always** on the phone with, he helps you
- [x] Manage Tasks
- [ ] Give reminders


Types of tasks currently  supported:
* Todo
* Deadline
* Event

Running it on a terminal
1. Download the jar file [here](https://github.com/hpkoh/ip/releases/tag/A-Jar)
2. Go to the directory where you saved the file
3. Run the command ```java -jar {filename}.jar```

This is how *Duke* class is ran 🏃 
```Java
public class Main {
    public static void main(String[] args) {
        new Duke("path/to/save/file").run();
    }
}
```

hpkoh committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

hpkoh committed:
A-Gradle

hpkoh committed:
Merge branch 'branch-A-Gradle'

hpkoh committed:
A-CheckStyle

hpkoh committed:
Merge branch 'branch-A-CheckStyle'

hpkoh committed:
Level-10

hpkoh committed:
Merge branch 'branch-Level-10'

hpkoh committed:
A-Assertions

Storage functions depending on filePath should not work with an empty filePath.
Task string representation should ensure that the object's status
matches whether it is done.

Storage class asserts that filePath is not empty.
Task class asserts that the valid correct symbol is used.

Assertions in Storage functions can check if the filePath field
is empty.
Assertion in Task toString can check if the object has the correct
status string.

hpkoh committed:
A-CodeQuality

Methods still supports unused text-base Ui
JavaDocs for many methods was barebones.
Code readibility can be further improved.

Deprecate text-based Ui functionalities and dependencies
Add JavaDocs to all public functions and non trivial private methods.
Refractor code to improve readibility

Refractoring code improves readibility and clarity of code thus allows
programme to be understood by developers working on it.

hpkoh committed:
A-CI

hpkoh committed:
Merge pull request #1 from hpkoh/branch-A-CodeQuality

A-CodeQuality

hpkoh committed:
Merge branch 'master' into branch-A-Assertions

hpkoh committed:
Merge pull request #2 from hpkoh/branch-A-Assertions

A-Assertions

hpkoh committed:
Merge branch 'master' into branch-A-CI

hpkoh committed:
Merge pull request #3 from hpkoh/branch-A-CI

A-CI

hpkoh committed:
C-Sort

Tasks are unsorted by default

Sorting on tasks allows user to see the tasks that are due the earliest
at the top.

Let's implement sort command to allow user to sort the task list
according to due date. Start date of event is taken as their due date.

Sorting by due date will allow users to see the most urgent tasks at the
top of the list.

hpkoh committed:
Merge branch 'branch-C-Sort'

hpkoh committed:
A-BetterGui

GUI is bland

Updated color scheme of messages

hpkoh committed:
A-UserGuide

No User Guide provided.

User guide teaches new users how to set up and use WingerIO easily.

User guide has been made using markdown.

hpkoh committed:
Set theme jekyll-theme-midnight

hpkoh committed:
Update README.md

hpkoh committed:
Rename Ui.png

hpkoh committed:
Update Ui.png
